lib/domain/orders_repository.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:mvp/classes/prefrenses.dart';
+import 'package:mvp/constants/apiCalls.dart';
+import 'package:mvp/domain/exceptions.dart';
+import 'package:mvp/models/ordersModel.dart';
+import 'package:http/http.dart' as http;
+
+abstract class OrdersRepository {
+  Future<List<OrderModel>> fetchOrders();
+}
+
+class OrdersRepositoryImpl implements OrdersRepository {
+  Future _fetch() async {
+    try {
+      final p = await Preferences.getInstance();
+      String id = await p.getData("id");
+      String token = await p.getData("token");
+      String url = APIService.ordersAPI + "$id";
+      Map<String, String> requestHeaders = {'x-auth-token': token};
+      var response = await http.get(url, headers: requestHeaders);
+      var responseJson = await _returnResponse(response);
+      return responseJson;
+    } on SocketException {
+      throw FetchDataException('No Internet Connection!');
+    }
+  }
+
+  @override
+  Future<List<OrderModel>> fetchOrders() async {
+    return await _fetch();
+  }
+
+  refreshToken() async {
```

Comment:
we have already implemented this function in a products bloc no ? why should we repeat it again ?